Does not build with generic-lens 2.0.0.0

Fails with:

```
src/Data/Generic/HKD/Named.hs:31:1: error:
    Could not load module ‘Data.GenericLens.Internal’
    It is a member of the hidden package ‘generic-lens-core-2.0.0.0’.
    Perhaps you need to add ‘generic-lens-core’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘generic-lens-1.1.0.0’.
    Perhaps you need to add ‘generic-lens’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
31 | import Data.GenericLens.Internal (GUpcast (..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

when including `generic-lens-core` it fails with:

```
src/Data/Generic/HKD/Named.hs:31:35: error:
    Module ‘Data.GenericLens.Internal’ does not export ‘GUpcast(..)’
   |
31 | import Data.GenericLens.Internal (GUpcast (..))
   |
```

It seems `GUpcast` has moved to https://hackage.haskell.org/package/generic-lens-core-2.0.0.0/docs/Data-Generics-Product-Internal-Subtype.html , however, it is no longer an exported module. When exporting it it builds.

I also had to import `HList` from `Data.Generics.Product.Internal.HList`.

I'll put up a PR with these changes, but I guess generic-lens would need to be updated to expose `GUpcast` again?

--- a/cabal.project
+++ b/cabal.project
@@ -1,9 +1,3 @@
 packages: .
 
 with-compiler: ghc-8.8.3
-
-source-repository-package
-  type: git
-  location: https://github.com/AlistairB/generic-lens.git
-  tag: e9fe04b049aa62c4489014537a4e2a0cf8644c79
-  subdir: generic-lens-core

--- a/higgledy.cabal
+++ b/higgledy.cabal
@@ -28,8 +28,8 @@ library
                    Data.Generic.HKD.Types
   build-depends: base >= 4.12 && < 5
                , barbies ^>= 2.0
-               , generic-lens ^>= 2.0
-               , generic-lens-core ^>= 2.0
+               , generic-lens >= 2.1 && < 3.0
+               , generic-lens-core >= 2.1 && < 3.0
                , QuickCheck >= 2.12.6 && < 2.14
                , named ^>= 0.3.0.0
   hs-source-dirs:      src

